### PR TITLE
[Resolve #288] Fix how user variables are accessed in config templating

### LINKED ISSUE #288: Unclear How to Access User Variables in Templating
Currently if i specify user variables as command line options:

`sceptre --var a=1 --var b=2 <command>`

The templating for both the environment and stack configs will receive the following dictionary: 

`{'var': {'user_variables': {'a': 1, 'b': 2}}}`

Is this expected behavior? Does this mean I can no longer access variables using `var.a`, instead now I need to write `var.user_variables.a`? If so then the documentation needs to be updated to reflect this.

### PR DESCRIPTION
**Pull request by cboss24**
Now variables supplied using the --var and --var-file command options
are available in the jinja templating as `{{ var.* }}`.

**Replace the text above the line with a summary of the changes in your PR.
The more detailed you are, the better. Before submitting the PR, make sure the following are checked.**

-----------------

* [x] Wrote [good commit messages][1].
* [x] [Squashed related commits together][2] after the changes have been approved.
* [x] Commit message starts with `[Resolve #issue-number]` (if a related issue exists).
* [x] Added unit tests.
* [ ] Added integration tests (if applicable).
* [x] All unit tests (`make test`) are passing.
* [x] Used the same coding conventions as the rest of the project.
* [x] The new code doesn't generate flake8 (`make lint`) offenses.
* [x] The PR relates to *only* one subject with a clear title
  and description in grammatically correct, complete sentences.

[1]: http://chris.beams.io/posts/git-commit/
[2]: https://github.com/todotxt/todo.txt-android/wiki/Squash-All-Commits-Related-to-a-Single-Issue-into-a-Single-Commit

### CODE REVIEW

**Comment by cboss24** [comment]
I was having a hard time unit testing `get_stack_or_env` and `get_stack` when the ConfigReader was being created internally. My solution was to break the ConfigReader initialization out into it's own function to be tested independently. Not sure if this is the best way, happy to hear your feedback.

**Comment by cboss24** [comment]
@seanrankine @b-t-g any thoughts on this?

File: sceptre/cli/helpers.py [hunk 1]
```diff
@@ -89,9 +89,20 @@ def write(var, output_format="str", no_colour=True):
     click.echo(stream)
 
 
+def _get_config_reader(ctx):
```
Comment (theseanything) [inline sceptre/cli/helpers.py]: So about to put a PR in that simplifies that CLI commands to just use the `get_stack_or_env` function. So what I suggest is you can probably remove this function and the tests and then rebase of the new master.

Comment (theseanything) [reply sceptre/cli/helpers.py]: #294

**Comment by cboss24** [comment]
Okay first round of requested changes complete.

File: tests/test_cli.py [hunk 2]
```diff
@@ -732,3 +733,27 @@ def test_CustomJsonEncoder_with_non_json_serialisable_object(self):
...
+        ctx = MagicMock(obj={
+            "sceptre_dir": sentinel.sceptre_dir,
+            "user_variables": sentinel.user_variables
+        })
+        stack, env = get_stack_or_env(ctx, "environment/dir")
+        self.mock_ConfigReader.assert_called_once_with(
+            sentinel.sceptre_dir, sentinel.user_variables
+        )
+        assert isinstance(env, Environment)
+        assert stack is None
```
Comment (theseanything) [inline tests/test_cli.py]: Could we add one more test to make sure `environment/dir/stack.yaml` returns a stack?

**Comment by cboss24** [comment]
Okay second round of requested changes complete.

**Pull request merged**